core/shared/src/main/scala/coursier/util/Parse.scala

```diff
@@ -164,29 +171,42 @@ object Parse {
 
     val attrs = strings.drop(1).map({ x => {
       if (x.mkString.contains(argSeparator)) {
-        throw new ModuleParseError(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
+        return Left(s"'$argSeparator' is not allowed in attribute '$x' in '$s'. Please follow the format " +
           s"'org${argSeparator}name[${argSeparator}version][${argSeparator}config]${attrSeparator}attr1=val1${attrSeparator}attr2=val2'")
       }
       val y = x.split("=")
       if (y.length != 2) {
-        throw new ModuleParseError(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
+        return Left(s"Failed to parse attribute '$x' in '$s'. Keyword argument expected such as 'classifier=tests'")
       }
       (y(0), y(1))
     }
     }).toMap
 
     val parts = coords.split(":", 5)
 
+    // Only "classifier" and "url" attributes are allowed
+    val validAttrsKeys = Set("classifier", "url")
+
+    validateAttributes(attrs, validAttrsKeys) match {
+      case Some(err) => return Left(err)
```

Comment:
I added this return statement here so that this error would behave the same way as the above errors do and stops the function from proceeding. If the return is not present, it doesn't stop the function and parsing will continue even though the attributes are invalid.